File: solutions/2766_find-the-prefix-common-array-of-two-arrays.py
# Problem: https://leetcode.com/problems/find-the-prefix-common-array-of-two-arrays
# Runtime: 3 ms

class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        seen = set()
        out = []
        n = 0

        for i in range(len(A)):
            if A[i] in seen: n += 1
            seen.add(A[i])
            if B[i] in seen: n += 1
            seen.add(B[i])

            out.append(n)

        return out

File: solutions/2766_find-the-prefix-common-array-of-two-arrays.java
// Problem: https://leetcode.com/problems/find-the-prefix-common-array-of-two-arrays
// Runtime: 1 ms

class Solution {
    public int[] findThePrefixCommonArray(int[] A, int[] B) {
        boolean[] seen = new boolean[A.length];
        int[] out = new int[A.length];
        int n = 0;

        for (int i = 0; i < A.length; i++) {
            if (seen[A[i] - 1])
                n += 1;
            seen[A[i] - 1] = true;

            if (seen[B[i] - 1])
                n += 1;
            seen[B[i] - 1] = true;

            out[i] = n;
        }

        return out;
    }
}